### DISCUSSION TITLE
Publishing markdown, with images, to Confluence

### DISCUSSION THREAD

**Discussion by angelotrivelli** [quarto-dev/quarto-cli#5144] [Q&A] [answered]
Super excited to see that it's possible to publish to Confluence. It works for me except for one huge show-stopper, images. 

Looking at the [docs under 'Content Limitations'](https://quarto.org/docs/publishing/confluence.html#content-limitations), it says:

> The confluence-html format supports nearly all of the standard Quarto markdown content types, including tables, callouts, and cross references.
>  
> However, there is currently no support for Citations, Videos, Diagrams, Tabsets, or Equations. In the future, we may add these features if there is a Confluence equivalent that can support the functionality.

It doesn't say anything about images?

I was hoping that `quarto publish confluence` would just upload the images I referenced in the markdown and somehow finagle the image url when it lands in confluence. *It does shows up in preview*, but sadly not in the confluence page.

Is there a clever way to get around this?

**Comment by dragonstyle** [comment, marked as answer]
You are actually correct - Confluence publishing should be uploading the images for you (it sometimes takes a minute or two for the image to available at the final URL in confluence, so you may see a placeholder for a minute or two the first time a page containing an image is deployed).

**Reply by angelotrivelli** [reply]
Oops! I was too impatient, it does work!

**Comment by ghost** [comment]
In practice, many folks are seeing this delay as a bug while Confluence is preparing the newly uploaded image attachments. 

I've created [this issue](https://github.com/quarto-dev/quarto-cli/issues/5151) to make this a better experience given our limitations.  We could provide better messaging in the CLI log potentially that may help.
